### ISSUE TITLE
Use the new Rustler "NIF features" to select the NIF version

### ISSUE THREAD

**Issue by philss** [philss/rustler-precompiled-action#3]
Starting in [v0.29](https://github.com/rusterlium/rustler/releases/tag/rustler-0.29.0), Rustler changed the way it selects the target NIF version to build. It now uses a set of Cargo features instead of the `RUSTLER_NIF_VERSION` env var.
So the env var is deprecated and is going to be removed in `v0.30`.

We need to support that, and maybe keep supporting the old way for a while.

What I imagine is to have a `features` attribute, that if present, we pass down to the `cargo / cross build` command.
We should probably keep the `nif-version` attribute, that should name the artifact properly.

**Comment by filmor** [comment]
Well, one could ask of library authors to use the same feature naming convention that `rustler` and `rustler_sys` use now. So `nif-version: 2.14` would translate to `cargo build --features nif_version_2_14`. You will need the `nif-version` parameter in any case to include the (claimed) NIF version level required for the compiled artifact. This could also be verified using something like `cargo tree -e features -p rustler -i --depth 1`.

**Comment by philss** [comment]
Yeah, makes sense! I was thinking of leave the feature name to the user, but using the same naming convention is a great idea.

I just didn't get exactly how I would verify this. I'm experimenting with Explorer, and my `Cargo.toml` looks like this:

```toml
[dependencies]
rustler = { version = "0.29", default-features = false, features = ["derive"] }

[features]
default = ["nif_version_2_15"]
nif_version_2_14 = ["rustler/nif_version_2_14"]
nif_version_2_15 = ["rustler/nif_version_2_15"]
nif_version_2_16 = ["rustler/nif_version_2_16"]
```

I know that for building, I would just do `cargo build --features nif_version_2_16` if I wanted to target NIF 2.16.
But how and when would I use `cargo tree -e features -p rustler -i --depth 1` to ensure that the correct feature was activated?

**Comment by filmor** [comment]
`cargo tree ...` shows you the features that are active on `rustler` (`rustler_sys` would do as well). You can pass the featues that you use for building as `--features` and then check, which nif version level is active on `rustler(_sys)`. Probably overkill, though.

**Comment by philss** [comment]
@filmor I think I'm covering the use cases here: https://github.com/philss/rustler-precompiled-action/pull/4

**Comment by philss** [comment]
This was released in version 1.1. I'm closing now. Thanks!
